packages/core/src/domain/configuration/configuration.ts

```diff
@@ -112,68 +86,50 @@ export function validateAndBuildConfiguration(
     return
   }
 
+  if (initConfiguration.sampleRate !== undefined && !isPercentage(initConfiguration.sampleRate)) {
+    display.error('Sample Rate should be a number between 0 and 100')
+    return
+  }
+
   // Set the experimental feature flags as early as possible so we can use them in most places
   updateExperimentalFeatures(initConfiguration.enableExperimentalFeatures)
 
-  const configuration: Configuration = {
-    beforeSend:
-      initConfiguration.beforeSend && catchUserErrors(initConfiguration.beforeSend, 'beforeSend threw an error:'),
-    cookieOptions: buildCookieOptions(initConfiguration),
-    service: initConfiguration.service,
+  return {
     ...computeTransportConfiguration(initConfiguration, buildEnv),
-    ...DEFAULT_CONFIGURATION,
-  }
-
-  if (initConfiguration.sampleRate !== undefined) {
-    if (!isPercentage(initConfiguration.sampleRate)) {
-      display.error('Sample Rate should be a number between 0 and 100')
-      return
-    }
-    configuration.sampleRate = initConfiguration.sampleRate
-  }
-
-  return configuration
-}
 
-export function buildConfiguration(initConfiguration: InitConfiguration, buildEnv: BuildEnv): Configuration {
-  const configuration: Configuration = {
     beforeSend:
       initConfiguration.beforeSend && catchUserErrors(initConfiguration.beforeSend, 'beforeSend threw an error:'),
     cookieOptions: buildCookieOptions(initConfiguration),
+    sampleRate: initConfiguration.sampleRate ?? 100,
     service: initConfiguration.service,
-    ...computeTransportConfiguration(initConfiguration, buildEnv),
-    ...DEFAULT_CONFIGURATION,
-  }
-
-  if ('allowedTracingOrigins' in initConfiguration) {
-    configuration.allowedTracingOrigins = initConfiguration.allowedTracingOrigins!
-  }
-
-  if ('sampleRate' in initConfiguration) {
-    configuration.sampleRate = initConfiguration.sampleRate!
+    silentMultipleInit: !!initConfiguration.silentMultipleInit,
+
+    /**
```

Comment:
Nit: I'd be in favour of adding comments on the type above. It will make it available for the completion

Reply:
But those comments are here to justify the values used, not to provide documentation on each key.. I think we should keep it close to the value.

Reply:
Fair enough. Lgtm then